src/analytics/gtag.js

```diff
@@ -0,0 +1,149 @@
+import getReadable from 'helpers/getReadable'
+import {
+  event,
+  getDatasetOptionsChanges,
+  getDatasetState,
+  getFormattedDatasetOptions,
+  getFilterCombination,
+  getParameterForLink,
+  getSnakeCase,
+  getToggledFilterItem
+} from './helpers'
+
+// NOTE: User-defined event parameters must be configured as custom definitions in the GA property.
+// Please make sure they are registered if you create any new ones.
+
+/* --- General --- */
+// tracks email subscriptions and the user-subscribed location
+const trackEmailSubscription = (Component) => {
+  const payload = {}
+  payload.subscription_click_from = Component.name
+  event('email_subscription', payload)
+}
+
+/* --- Compendia --- */
+// tracks the compendia downloads and organism names
+const trackCompendiaDownload = (compendia) => {
+  const key = 'quant_sf_only'
+  const type = getReadable(key, compendia[key])
+  const payload = {}
+  payload[`${type}_organism`] = compendia.primary_organism_name
+  event(`compendia_${type}_download`, payload)
+}
+
+/* --- Datasets --- */
+// tracks user clicks on the dataset actions buttons
+const trackDatasetAction = (Component) => {
+  const payload = {}
+  payload.my_dataset_action = Component.name
+  event('my_dataset_action', payload)
+}
+// tracks the dataset download options selected by the user
+const tracktrackDatasetDownloadOptions = (dataset) => {
+  const payload = {}
+  payload.my_dataset_download_options = getFormattedDatasetOptions(dataset)
+  event('my_dataset_download_options', payload)
+}
+// tracks the number of dataset downloads associated with each token
+const trackDatasetDownload = (token) => {
+  const payload = {}
+  payload.token = token
+  event('dataset_downalod', token)
+}
```

Comment:
We don't need the token for this. Really shouldn't be sending that out.

Reply:
I originally borrowed the concept from the current `Refine.bio - GA4` in google analytics, which tracks API token creations by counting unique `token` values. I initially adapted this idea to count unique user dataset downloads. 

I've removed the `token` parameter and `trackDatasetDownload` now simply counts the dataset download as `dataset_download`, see at 3c77006020fef5ed3bef96e14baed8192e7d0743 👍